lib/matplotlib/backends/backend_qt5agg.py

```diff
@@ -38,16 +38,19 @@ def paintEvent(self, event):
 
         painter = QtGui.QPainter(self)
 
-        rect = event.rect()
-        left = rect.left()
-        top = rect.top()
-        width = rect.width()
-        height = rect.height()
         # See documentation of QRect: bottom() and right() are off by 1, so use
         # left() + width() and top() + height().
-        bbox = Bbox(
-            [[left, self.renderer.height - (top + height * self._dpi_ratio)],
-             [left + width * self._dpi_ratio, self.renderer.height - top]])
+        rect = event.rect()
+        # scale rect dimensions using the screen dpi ratio to get
+        # correct values for the Figure coordinates (rather than QT5's coords)
+        width = rect.width() * self._dpi_ratio
+        height = rect.height() * self._dpi_ratio
+        left, top = self.mouseEventCoords(rect.topLeft())
+        # shift the "top" by the height of the image to get the
+        # correct corner for our coordinate system
+        top -= height
```

Comment:
I guess the point is that this is really bottomRight(), but with the off-by-1 correction? perhaps move the comment about the off-by-1 there, if I got this correctly.